DrawerKit/DrawerKit/Internal API/AnimationController.swift

```diff
@@ -2,50 +2,131 @@ import UIKit
 
 final class AnimationController: NSObject {
     private let isPresentation: Bool
-    private let durationInSeconds: TimeInterval
-    private let timingCurveProvider: UITimingCurveProvider
+    private let configuration: DrawerConfiguration
 
-    init(isPresentation: Bool,
-         durationInSeconds: TimeInterval,
-         timingCurveProvider: UITimingCurveProvider) {
+    init(isPresentation: Bool, configuration: DrawerConfiguration) {
         self.isPresentation = isPresentation
-        self.durationInSeconds = durationInSeconds
-        self.timingCurveProvider = timingCurveProvider
+        self.configuration = configuration
         super.init()
     }
 }
 
 extension AnimationController: UIViewControllerAnimatedTransitioning {
     func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
-        return durationInSeconds
+        guard let context = transitionContext else { return 0 }
+        let (_, presentedVC) = viewControllers(context, isPresentation)
+        return actualTransitionDuration(presentedVC, context)
     }
 
     func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
-        let key = (isPresentation ?
-            UITransitionContextViewControllerKey.to :
-            UITransitionContextViewControllerKey.from)
+        let (presentingVC, presentedVC) = viewControllers(transitionContext, isPresentation)
+        if isPresentation { transitionContext.containerView.addSubview(presentedVC.view) }
 
-        let controller = transitionContext.viewController(forKey: key)!
-        if isPresentation { transitionContext.containerView.addSubview(controller.view) }
+        let duration = transitionDuration(using: transitionContext)
+        let timingCurveProvider = configuration.timingCurveProvider
+        let animator = UIViewPropertyAnimator(duration: duration, timingParameters: timingCurveProvider)
 
-        let presentedFrame = transitionContext.finalFrame(for: controller)
-        var dismissedFrame = presentedFrame
-        dismissedFrame.origin.y = transitionContext.containerView.frame.size.height
+        let (initialFrame, finalFrame) = frames(presentedVC, transitionContext)
 
-        let initialFrame = (isPresentation ? dismissedFrame : presentedFrame)
-        let finalFrame = (isPresentation ? presentedFrame : dismissedFrame)
+        let containerBounds = transitionContext.containerView.bounds
+        let containerViewH = containerBounds.size.height
 
```

Comment:
`containerHeight` would or `containerH` is more consistent with the naming of other variable in this method.